### PR TITLE
adds kaggle dataset

### PR DESCRIPTION
**Pull request by aditya-essential**
Adds kaggle dataset

### CODE REVIEW

File: ft_datasets/kaggle_dataset/kaggle_dataset.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        self.tokenizer = tokenizer
+
+    def __len__(self):
+        return len(self.data)
+
+    def __getitem__(self, index):
+        data = self.data[index]
+        prompt = data["prompt"]
+        example = prompt + data["response"]
+        prompt = torch.tensor(
```
Comment (nikiparmar) [inline ft_datasets/kaggle_dataset/kaggle_dataset.py:29]: can you add a comment on what do prompt, example and response mean here? (what is trained , what is input ,etc)

File: ft_datasets/kaggle_dataset/kaggle_dataset.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+from typing import List
+
+
+class InstructionDataset(Dataset):
+    def __init__(self, dataset_config, tokenizer, partition="train", max_words=4000):
+        self.data = []
+        with open(dataset_config.data_path) as f:
+            self.data = [json.loads(line) for line in f]
+        if partition == "train":
+            self.data = self.data[500:15000]
```
Comment (nikiparmar) [inline ft_datasets/kaggle_dataset/kaggle_dataset.py:16]: add constants for num_train and num_eval examples?

File: ft_datasets/kaggle_dataset/kaggle_dataset.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+        # Calculate padding.
+        padding = self.max_words - example.shape[0]
+        # Padding
+        if padding > 0:
+            example = torch.cat((example, torch.zeros(padding, dtype=torch.int64) - 1))
+            labels = copy.deepcopy(example)
+            clipping = max(0, len(prompt)-1)
+            # Mask the input labels.
+            labels[:clipping] = -1
```
Comment (nikiparmar) [inline ft_datasets/kaggle_dataset/kaggle_dataset.py:47]: add a class variable for the mask token value -1?

File: ft_datasets/kaggle_dataset/kaggle_dataset.py [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+class InstructionDataset(Dataset):
+    def __init__(self, dataset_config, tokenizer, partition="train", max_words=4000):
+        self.data = []
+        with open(dataset_config.data_path) as f:
+            self.data = [json.loads(line) for line in f]
+        if partition == "train":
+            self.data = self.data[500:15000]
+        else:
+            self.data = self.data[:500]
+        self.max_words = max_words
```
Comment (nikiparmar) [inline ft_datasets/kaggle_dataset/kaggle_dataset.py:19]: Is it max_words or max_tokens?

File: ft_datasets/kaggle_dataset/kaggle_dataset.py [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+            example = example[-self.max_words:]
+            labels = copy.deepcopy(example)
+            # Mask the input labels.
+            if self.max_words-response_len > 0:
+                labels[: self.max_words-response_len] = -1 
+        
+        example_mask = example.ge(0)
+        label_mask = labels.ge(0)
+        example[~example_mask] = 0
+        labels[~label_mask] = -100
```
Comment (nikiparmar) [inline ft_datasets/kaggle_dataset/kaggle_dataset.py:60]: make this a constant (-100)

File: ft_datasets/kaggle_dataset/kaggle_dataset.py [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+            response_len = response.shape[0]
+            example = example[-self.max_words:]
+            labels = copy.deepcopy(example)
+            # Mask the input labels.
+            if self.max_words-response_len > 0:
+                labels[: self.max_words-response_len] = -1 
+        
+        example_mask = example.ge(0)
+        label_mask = labels.ge(0)
+        example[~example_mask] = 0
```
Comment (nikiparmar) [inline ft_datasets/kaggle_dataset/kaggle_dataset.py:59]: is this statement needed?

Comment (nikiparmar) [reply ft_datasets/kaggle_dataset/kaggle_dataset.py:59]: Just to clarify - input pad tokens will stay 0, but the labels will be updated to -100 to be used during loss calculation?

Comment (nikiparmar) [reply ft_datasets/kaggle_dataset/kaggle_dataset.py:59]: Discussed offline, but add a comment for the example mask vs label mask

**Review by vaswani — COMMENTED** [review]
Higher level comment. could you link to the wandb as part of the PR ?

**Review by vaswani — COMMENTED** [review]
some quick comments. still reviewing.

File: ft_datasets/kaggle_dataset/kaggle_dataset.py [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+            self.tokenizer.encode(data["response"]), dtype=torch.int64
+        )
+
+        # Calculate padding.
+        padding = self.max_words - example.shape[0]
+        # Padding
+        if padding > 0:
+            example = torch.cat((example, torch.zeros(padding, dtype=torch.int64) - 1))
+            labels = copy.deepcopy(example)
+            clipping = max(0, len(prompt)-1)
```
[the remark this answers is not in the document]
Comment (nikiparmar) [reply ft_datasets/kaggle_dataset/kaggle_dataset.py:51]: Can you add a comment for this logic from our discussion yesterday

**Review by nikiparmar — APPROVED** [review]
lgtm!
